typed-protocols/src/Network/TypedProtocol/Proofs.hs

```diff
@@ -0,0 +1,184 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE TypeInType #-}
+
+
```

Comment:
I recommend flipping on `-fwarn-incomplete-patterns` with a pragma. Very important for proofs.

Reply:
Is that not already included in `-Wall`?

Reply:
`-Wall` isn't flipped on here either.

Reply:
Done.